File: models/post_tag.py
#!/usr/bin/python3
"""Association table for posts and tags"""
from sqlalchemy import Column, ForeignKey, String, Table
from models.base_model import BaseModel, Base

post_tag = Table(
    'post_tags',
    Base.metadata,
    Column('post_id', String(60), ForeignKey('posts.id'), primary_key=True),
    Column('tag_id', String(60), ForeignKey('tags.id'), primary_key=True),
)

File: api/models/post_tag.go
package models

import (
	"github.com/google/uuid"
)

// PostTag represents the many-to-many relationship between posts and tags.
type PostTag struct {
	PostID uuid.UUID `gorm:"type:char(36);primaryKey" json:"post_id"`
	TagID  uuid.UUID `gorm:"type:char(36);primaryKey" json:"tag_id"`
}

// TableName overrides the default table name for the join table.
func (PostTag) TableName() string {
	return "post_tags"
}
